build.sh

```diff
@@ -1,14 +1,38 @@
 #!/bin/sh
+set -e
+
+if [ -z "$CC" ]; then
+    CC=gcc
+fi
+
 rm NAR
 rm src/RuleTable.c
-set -e
-Str=`ls src/*.c src/NetworkNAR/*.c | xargs`
-echo $Str
+sources=`ls src/*.c src/NetworkNAR/*.c | xargs`
+echo $sources
 echo "Compilation started:"
-BaseFlags="-mfpmath=sse -msse2 -pthread -lpthread -D_POSIX_C_SOURCE=199506L -pedantic -std=c99 -g3 -O3 $Str -lm -oNAR"
+BaseFlags="-pthread -lpthread -D_POSIX_C_SOURCE=199506L -pedantic -std=c99 -g3 -O3 $sources -lm -oNAR"
 NoWarn="-Wno-tautological-compare -Wno-dollar-in-identifier-extension -Wno-unused-parameter -Wno-unused-variable"
-gcc -DSTAGE=1 -Wall -Wextra -Wformat-security $NoWarn $BaseFlags
+
+# check if we are on Windows, use SSE2 then
+if [ "$OSTYPE" == "cygwin" ]; then
+    echo "Compiling for Windows (x86/x64)."
+    # CC=i686-w64-mingw32-gcc
+    BaseFlags="-mfpmath=sse -msse2 $BaseFlags"
+fi
+
+# if CPU architecture is x86/x64, then use SSE2
+if [[ "$OSTYPE" != "cygwin" && "$(uname -m)" == "x86"* ]]; then
```

Comment:
You cannot use `[[` in /bin/sh as it is an enhanced keyword provided by bash, which /bin/sh does not have. You will need to use the single `[` version, which also means you cannot use `&&` inside the test -- instead use
```suggestion
if [ "$OSTYPE" != "cygwin" ] && [ "$(uname -m)" = "x86"* ]]; then
```